File: app/src/main/java/com/blues/gankio/_new/mvp/view/GankViewHolder.java
package com.blues.gankio._new.mvp.view;

import android.app.Activity;
import android.content.Intent;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.blues.R;
import com.blues.gankio._new.detail.GankPostDetailActivity;
import com.blues.gankio._new.mvp.model.GankUniversalEntity;
import com.bumptech.glide.Glide;
import com.jude.easyrecyclerview.adapter.BaseViewHolder;

import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.core.app.ActivityOptionsCompat;

public class GankViewHolder extends BaseViewHolder<GankUniversalEntity.DataBean> {

    private ImageView ivPreview;
    private TextView tvTitle, tvAuthor, tvViews, tvLikes, tvPTime;
    private ConstraintLayout clRoot;

    public GankViewHolder(ViewGroup parent) {
        super(parent, R.layout.item_gank);

        tvTitle = $(R.id.tv_title);
        tvAuthor = $(R.id.tv_author);
        tvViews = $(R.id.tv_views);
        tvLikes = $(R.id.tv_likes);
        tvPTime = $(R.id.tv_publish_time);
        ivPreview = $(R.id.iv_preview);
        clRoot = $(R.id.cl_root);
    }

    @Override
    public void setData(GankUniversalEntity.DataBean data) {
        super.setData(data);

        tvTitle.setText(data.getTitle());
        tvAuthor.setText(String.format("作者：%s", data.getAuthor()));

        if (data.getImages().size() != 0) {
            Glide.with(getContext()).load(data.getImages().get(0)).placeholder(R.mipmap.ic_img_error).into(ivPreview);
        } else {
            Glide.with(getContext()).load(R.mipmap.ic_img_error).placeholder(R.mipmap.ic_img_error).into(ivPreview);
        }

        tvViews.setText(String.valueOf(data.getViews()));
        tvLikes.setText(String.valueOf(data.getLikeCounts()));
        tvPTime.setText(String.format("日期:%s", data.getPublishedAt()));

        clRoot.setOnClickListener(v -> {
            Intent intent = new Intent(getContext(), GankPostDetailActivity.class);
            intent.putExtra("postId", data.get_id());
            ActivityOptionsCompat options = ActivityOptionsCompat.makeSceneTransitionAnimation((Activity) getContext(), clRoot, "transitionGank");
            getContext().startActivity(intent, options.toBundle());
        });
    }
}

File: app/src/main/java/com/blues/gankio/v2/view/GankViewHolder.kt
package com.blues.gankio.v2.view

import android.view.ViewGroup
import com.blues.gankio.v2.model.GankioUniversalBean
import com.blues.R
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import android.content.Intent
import androidx.core.app.ActivityOptionsCompat
import android.app.Activity
import android.widget.ImageView
import coil.load
import com.jude.easyrecyclerview.adapter.BaseViewHolder

class GankViewHolder(parent: ViewGroup?) :
    BaseViewHolder<GankioUniversalBean.DataBean>(parent, R.layout.item_gank) {

    private val ivPreview: ImageView = itemView.findViewById(R.id.iv_preview)
    private val tvTitle: TextView = itemView.findViewById(R.id.tv_title)
    private val tvAuthor: TextView = itemView.findViewById(R.id.tv_author)
    private val tvViews: TextView = itemView.findViewById(R.id.tv_views)
    private val tvLikes: TextView = itemView.findViewById(R.id.tv_likes)
    private val tvPTime: TextView = itemView.findViewById(R.id.tv_publish_time)
    private val clRoot: ConstraintLayout = itemView.findViewById(R.id.cl_root)

    override fun setData(data: GankioUniversalBean.DataBean) {
        super.setData(data)
        tvTitle.text = data.title
        tvAuthor.text = "作者：${data.author}"
        if (data.images.isNotEmpty()) {
            ivPreview.load(data.images[0]){
                placeholder(R.mipmap.ic_img_error)
                error(R.mipmap.ic_img_error)
            }
        } else {
            ivPreview.load(R.mipmap.ic_img_error)
        }
        tvViews.text = data.views.toString()
        tvLikes.text = data.likeCounts.toString()
        tvPTime.text = "日期:${data.publishedAt}"
        clRoot.setOnClickListener {
            val intent = Intent(context, GankPostDetailActivity::class.java).putExtra("postId", data.id)
            val options = ActivityOptionsCompat.makeSceneTransitionAnimation((context as Activity), clRoot, "transitionGank")
            context.startActivity(intent, options.toBundle())
        }
    }
}